Mathlib/Algebra/Category/ModuleCat/OverFilteredColimits.lean

```diff
@@ -0,0 +1,351 @@
+    smulColimit r (m₁ + m₂) = smulColimit r m₁ + smulColimit r m₂ := by
+  classical
+  let O : Finset J :=
+    { c, Concrete.indexRepColimit ℳ m₁, Concrete.indexRepColimit ℳ m₂ }
+  let H : Finset ((X : J) ×' (Y : J) ×' (_ : X ∈ O) ×' (_ : Y ∈ O) ×' (X ⟶ Y)) := {}
+  let j : J := IsFiltered.sup O H
+
+  have eq₁ : m₁ = colimit.ι ℳ j
+      (ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₁)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₂ : m₂ = colimit.ι ℳ j
+      (ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₂)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₃ : m₁ + m₂ = colimit.ι ℳ j
+      (ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₁) +
+       ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₂)) := by
+    rw [map_add]
+    erw [colimit.w_apply, colimit.w_apply, Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+
+  rw [eq₃]
+  conv_rhs => rw [eq₁]; rhs; rw [eq₂]
+  rw [smul_rep, smul_rep, smul_rep, hsmul.smul_add, map_add]
+
+lemma add_smul (c : J) (r₁ r₂ : ℛ.obj c) (m : colimit (C := 𝔄𝔟) ℳ) :
+    smulColimit (r₁ + r₂) m = smulColimit r₁ m + smulColimit r₂ m := by
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smul_rep, hsmul.add_smul, smul_rep, smul_rep, map_add]
+
+lemma zero_smul (c : J) (m : colimit (C := 𝔄𝔟) ℳ) :
+    smulColimit (ℳ := ℳ) (0 : ℛ.obj c) m = 0 := by
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smul_rep, hsmul.zero_smul, map_zero]
+
+end smulColimit
+
+noncomputable instance moduleObjColimit (j : J) :
+    Module (ℛ.obj j) (colimit (C := 𝔄𝔟) ℳ) where
+  smul := smulColimit
+  one_smul := smulColimit.one_smul _ _ _
+  mul_smul := smulColimit.mul_smul _ _ _
+  smul_zero := smulColimit.smul_zero _ _ _
+  smul_add := smulColimit.smul_add _ _ _
+  add_smul := smulColimit.add_smul _ _ _
+  zero_smul := smulColimit.zero_smul _ _ _
+
+variable {ℛ ℳ} in
+/--
+Let `R` be the filtered colimit of rings `{Rⱼ}` and `M` be the filtered colimit of
+abelian groups `{Mⱼ}`  with the same indexing set `j ∈ J`, if for each `j ∈ J`, `Mⱼ` is an `Rⱼ` such
+that the `Rⱼ`-action is compatible, then there is a scalar multiplication
+`R → M → M`.
+-/
+noncomputable def colimitSMulColimit (r : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimit (C := 𝔄𝔟) ℳ :=
+  (smulColimit (Concrete.repColimit ℛ r) m)
+
+namespace colimitSMulColimit
+
+lemma rep_smul {c : J} (r : ℛ.obj c) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (colimit.ι ℛ c r) m = smulColimit r m := by
+  delta colimitSMulColimit
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smulColimit.smul_rep]
+  apply hsmul.respect_ι
+  · erw [Concrete.ι_repColimit_eq]
+  · erw [Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
```

Comment:
```suggestion
    rw [Concrete.ι_repColimit_eq], smulColimit.smul_rep]
  apply hsmul.respect_ι
  · rw [Concrete.ι_repColimit_eq]
  · rw [Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
```
At this point in the review I was wondering why the heck you needed so many `erw`s so I removed a few and it turns out that you don't need them at all. How many other `erw`s should be `rw`?